src/connectors/impls/file.rs

```diff
@@ -41,45 +45,24 @@ pub(crate) enum Mode {
     Overwrite,
 }
 
-impl Mode {
-    fn as_open_options(&self) -> OpenOptions {
-        let mut o = OpenOptions::new();
-        match self {
-            Self::Read => {
-                o.read(true);
-            }
-            Self::Append => {
-                o.create(true).write(true).append(true);
-            }
-            Self::Write | Self::Truncate => {
-                o.create(true).write(true).truncate(true);
-            }
-            Self::Overwrite => {
-                o.create(true).write(true);
-            }
-        }
-        o
-    }
-}
-
 /// File connector config
 #[derive(Deserialize, Debug, Clone)]
 #[serde(deny_unknown_fields)]
 pub(crate) struct Config {
-    /// path to the file
-    pub(crate) path: PathBuf,
-    /// how to interface with the file
-    pub(crate) mode: Mode, // whether we read or write (in various forms)
     /// chunk_size to read from the file
     #[serde(default = "default_buf_size")]
     pub(crate) chunk_size: usize,
+
+    pub(crate) initial_command: Option<crate::connectors::traits::Command>,
```

Comment:
I don't know, standarisation is definitely good, OTOH not all connectors will be command driven. It would be nice to be able to make that an option for all the command driven connectors, and only those, but I don't think that's easily doable with the current abstractions that we have

Reply:
Verify means that also other errors could be caught at "compile time" which would be really nice to be hoenst. Things like malformated commands, or commands that are not supported